### ISSUE TITLE
Drush unable to uninstall modules with missing tables

### ISSUE THREAD

**Issue by bkline** [drush-ops/drush#4531]
**Describe the bug**
When a `drush pm:enable` command fails, it leaves the database in a broken state. You can't fix the problem which caused the failure and reinstall (drush says the module is already installed and does nothing), you can't uninstall the module (drush complains that it can't find one or more tables that it wants to remove), you can't update the database with `drush updatedb` (drush reports that there are no pending updates).

**To Reproduce**
1. Install a fresh Drupal 9.x
2. Create a custom content entity module
3. Add a typo to the entity creation file (misplaced semicolon, for example)
4. Run `drush pm:enable modulename`
5. Note failure
6. Fix typo
7.  Run `drush pm:enable modulename`
8. Confirm that `drush` did nothing
9. Run `drush pm-uninstall modulename`
10. Note failure

**Expected behavior**
I expect the `drush` will either leave the module in its disabled state when the enable request fails, or (if for some reason that is not possible) recognize, when it is asked to uninstall the module, that the table it wants to delete is already gone and proceed with a warning (and/or a request for confirmation if there are cases in which proceeding is unsafe).

**Actual behavior**
`Drush` leaves the database in a broken state.

**Workaround**
The only way out of this situation is to trick `drush` by creating a dummy table that it can delete. `Drush` doesn't realize or care what the table structure looks like, it just wants the satisfaction of knowing it has deleted the table. `CREATE TABLE tablename (whatever INT)` will do.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 10.3.2
| Drupal version? | 9.0.3
| PHP version?     | 7.4.3
| OS?             | Linux

**Additional information**
I'm not the only one who's run into this condition. It looks very much like [this ticket](https://www.drupal.org/project/drupal/issues/3099084) filed against Drupal core was dealing with the same problem (at several places in that ticket's comments the core maintainers pointed out that the failures being reported there looked more like a problem with `drush` than with Drupal core).

**Comment by weitzman** [comment]
That ticket is very long. I'm not seeing references to pm:enable command.

As you can see at https://github.com/drush-ops/drush/blob/d8234e57007693686d077826579a9671f620301d/src/Drupal/Commands/pm/PmCommands.php#L106-L120, pm:uninstall command is just a thin wrapper around Drupal core's ModuleInstaller service. So any cleanup or misbehavior is better handled there. At least try your experiment with Drupal core UI and see if it behaves any better.

**Comment by bkline** [comment]
Cool! So the Drush project points the finger at core, and the core team points the finger at Drush. 😂

>  I'm not seeing references to pm:enable command.

No, but you will find `drush pm:uninstall` (the command whose failure is the subject of this ticket), followed by a report of the same error message I'm getting.

I understand what you're saying, and I'll shine the flashlight back on the core code. 🔦

**Comment by bkline** [comment]
After some digging (mostly wading through lots of reports of the same problem) I found this [five-year-old bug report](https://www.drupal.org/project/drupal/issues/2563515). Still open, and it appears that -- after a three-year hiatus while the bug (with patches) was ignored -- they've come back to it, only to realize that they aren't happy their decisions as to how fix the bug. 😔

In any case, I agree that the ball's not in your court. Thanks for your speedy response. The fact that you took the trouble to point to the core code `drush` is invoking was what made it possible for me to find the upstream bug report. Closing this ticket.
